### DISCUSSION TITLE
Changing settings.yml without running install_and_upgrade.sh all the way

### DISCUSSION THREAD

**Discussion by airlomba** [jekhokie/raspberry-noaa-v2#425] [Ideas]
Hello all,
I have been struggling a bit to get this tool running in a Raspberry Pi B+ version 1.2. It runs! Slowly but it runs (except for METEOR that need more RAM than available in the Pi).
The worst parts in adjusting the settings are:
- waiting for another satellite pass to get some results (nobody's fault)
- and the 10 minutes run of the script install_and_upgrade.sh that, I believe, does 95% useless work if the only change is in the settings.yml file.
Can anyone point me some direction in order to save some time? Or, if there isn't any ready solution, may be this can be a suggestion for an improvement. Anyway, thank you for your time and work.
Best regards.

**Comment by dom-robinson** [comment]
I have to agree that the the current updating process using Ansible is prooving a hammer to crack a nut. In some use cases it makes sense but in broad terms I am of the opinion that without @jekhokie's ninja skillz with Ansible we are all finding it hard to add modifications and updates. I know @colinluthier shares this view too. 

@Cadair has more skills than Colin and I in this specific area, where Colin and I are typically a little more focussed on function and feature. The upside of the Ansible approach is that it standardised the approach for all of us and avoids (for example) my 'frankencode' going into production in a working but hacky way - which at first is fun, but then becomes a nightmare.

That said the effort to 'de-ansible-ise' the current system is not inconsiderable, and would then open up a number of issues such as platform variation support which means that while it may be slow running on your PI B+ etc it DOES actually run, since hte ansible playbook makes sure all the dependancies are in, and if we remove that checking then almost every user would have a unique setup and need to sort thier own deps manually (resulting in pain and frustration or support requests!!)


Its a fairly big existential question IMHO.

For now i want to try to get the current round of key issues worked in as is, and then I think we need to look at how we can update the ansible to more quickly 'skip' things that don't need to be re-done every time the update is run. Note that on my fairly recent PI the install and update takes about 2 mins which i find acceptible.. 

Anyway - hope that helps a bit. And thanks for the input...

**Reply by airlomba** [reply]
Hi Dom,
Thank you for your answer. I now have a feeling of the virtues of Ansible. However, it is not my intention to get rid of Ansible. I think my previous message was not written the way it should have been. I was just asking for some kind of shortcut to get the system to consider the new settings.yml without having to check all the other stuff over again. For exemple, in the case of having just tweaked the gain for NOAA15...
Anyway, I understand there is a lot more to improve than just try to save some time. :) Also, in the future I will probably replace this Pi with one of a recent generation. I just tried to use the V1.2 as a challenge to see if the system would work on such early Pi; and it works! So, good job guys! Thank you all.
Regards.

**Comment by RyLeeRyno** [comment]
you can edit the install scripts config output directly here:  ~/.noaa-v2.conf
